src/hotspot/cpu/ppc/c2_MacroAssembler_ppc.cpp

```diff
@@ -626,3 +626,72 @@ void C2_MacroAssembler::count_positives(Register src, Register cnt, Register res
   bind(Ldone);
   subf(result, src, result);      // Result is offset from src.
 }
+
+void C2_MacroAssembler::reduceI(int opcode, Register dst, Register iSrc, VectorRegister vSrc,
+                                VectorRegister vTmp1, VectorRegister vTmp2) {
+
+  auto fn_vec_op = [this](int opcode, const VectorRegister &dst, const VectorRegister &a, const VectorRegister &b) {
+    switch(opcode) {
+      case Op_AddReductionVI:
+        vadduwm(dst, a, b);
+        break;
+      case Op_MulReductionVI:
+        vmuluwm(dst, a , b);
+        break;
+      case Op_AndReductionV:
+        vand(dst, a, b);
+        break;
+      case Op_OrReductionV:
+        vor(dst, a, b);
+        break;
+      case Op_XorReductionV:
+        vxor(dst, a, b);
+        break;
+      case Op_MinReductionV:
+        vminsw(dst, a, b);
+        break;
+      case Op_MaxReductionV:
+        vmaxsw(dst, a, b);
+        break;
+      default: assert(false, "wrong opcode");
```

Comment:
Maybe put each case in one line? Would be more compact. Up to you.

Reply:
Good idea. Looks better when compacted.